pages/series/[slug].vue

```diff
@@ -0,0 +1,329 @@
+<script setup>
+// COMPONENT RE-IMPORTS
+// TODO: remove when we have implemented component library as a module
+// https://nuxt.com/docs/guide/directory-structure/components#library-authors
+import { BlockEventDetail, BlockInfo, BlockTag, CardMeta, DividerWayFinder, FlexibleMediaGalleryNewLightbox, NavBreadcrumb, ResponsiveImage, RichText, SectionScreeningDetails, SectionTeaserCard, SectionWrapper } from 'ucla-library-website-components'
+
+// HELPERS
+import _get from 'lodash/get'
+
+// GQL
+import FTVAEventSeriesDetail from '../gql/queries/FTVAEventSeriesDetail.gql'
+
+const { $graphql } = useNuxtApp()
+
+const route = useRoute()
+
+// DATA
+const { data, error } = await useAsyncData(`ftva-event-series-detail-${route.params.slug}`, async () => {
+  const data = await $graphql.default.request(FTVAEventSeriesDetail, { slug: route.params.slug })
+  return data
+})
+if (error.value) {
+  throw createError({
+    ...error.value, statusMessage: 'Page not found .' + error.value, fatal: true
+  })
+}
+
+if (!data.value.ftvaEventSeries) {
+  throw createError({
+    statusCode: 404,
+    statusMessage: 'Page Not Found for ftvaEventSeries',
+    fatal: true
+  })
+}
+
+const page = ref(_get(data.value, 'ftvaEventSeries', {}))
+const upcomingEvents = ref(_get(data.value, 'upcomingEvents', {}))
+const pastEvents = ref(_get(data.value, 'pastEvents', {}))
+const otherSeriesOngoing = ref(_get(data.value, 'otherSeriesOngoing', {}))
+const otherSeriesUpcoming = ref(_get(data.value, 'otherSeriesUpcoming', {}))
+
+watch(data, (newVal, oldVal) => {
+  console.log('In watch preview enabled, newVal, oldVal', newVal, oldVal)
+  page.value = _get(data.value, 'ftvaEventSeries', {})
+  upcomingEvents.value = ref(_get(data.value, 'upcomingEvents', {}))
+  pastEvents.value = ref(_get(data.value, 'pastEvents', {}))
+  otherSeriesOngoing.value = ref(_get(data.value, 'otherSeriesOngoing', {}))
+  otherSeriesUpcoming.value = ref(_get(data.value, 'otherSeriesUpcoming', {}))
+})
+
+// Get data for Image or Carousel at top of page
+const parsedImage = computed(() => {
+  return page.value.imageCarousel
+})
+
+// Transform data for Carousel
+const parsedCarouselData = computed(() => {
+  // map image to item, map creditText to credit
+  return parsedImage.value.map((rawItem, index) => {
+    return {
+      item: [{ ...rawItem.image[0], kind: 'image' }], // Carousels on this page are always images, no videos
+      credit: rawItem?.creditText,
+      captionTitle: 'dfdsfs', // TODO do we need these? test without
+      captionText: 'dfsdfsd',
+    }
+  })
+})
+
+const parsedFtvaEventSeries = computed(() => {
+  /* Early Returns: If series.value is falsy or empty, or
+    if firstSeries.ftvaEvent is falsy or empty, return
+    an empty array immediately. */
+  if (!series.value || series.value.length === 0) {
```

Comment:
Remove this computed method as there is no series variable in this file. and logic for explore more series is going to be different